src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -872,19 +872,23 @@ export class GuidedTourService {
     /**
      *  @return true if highlighted element is available, otherwise false
      */
-    private checkSelectorValidity(): boolean {
+    private checkSelectorValidity(): boolean | void {
```

Comment:
Why do we change the signature here? Can we not just `return false;` after resetting the tour?

Reply:
tried that, after returning false, it has some strange behavior. It assumes you completed the tour, so and displays congratulation pop-up :) it's confusing, dunno

Reply:
Ok. Does it misbehave when returning true as well? Fwiw we should probably not reset the tour from inside this method then?

Reply:
Actual problem is higlightselector is undefined in the sentry. I tried to debug for user who is not part of tutor of tutorial course. The `currentTour.steps` is empty in here, so currentTourStep is also undefined. I added simple checks for that. After fixing that, it finds other error I mentioned Pr description. After fixing last error, it just begins to displays this "Congrats for completing this tour' popup to user who is not part of tutor of tutorial course. So, it just does not make sense. I mentioned this to Simon because he is familiar with this tour concept before. We thought that resetting the tour in here could resolve that., cuz we encountered the issue in here at first. However, I am open other ideas as well.

Reply:
Hm, ok. I don't have a better idea right now so we might as well leave it as is then.

Reply:
@FrankeLukas 👍 
I think returning `bool | void` is not the right approach here.

After writing a small example in the typescript sandbox, I figured out that `void` case is equivalent to `false`.
Therefore it's better to return `false` rather than returning nothing.

Link to the sandbox:
https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMAFADwFyLCAtgIwFMAnASkS3zjgBtCBDJAH0QDc4YATRAbwChEgxDGCJ0iALxTEARnL8hixMUJQQxJFGIhCAbgFCAvn2N9W9YojSTkKAEzkA-IgDMFRABZ9EBAGdahAB0NHAA5uikukA